pkg/receive/handler.go

```diff
@@ -159,14 +158,12 @@ func (h *Handler) Run(ctx context.Context) error {
 		ErrorLog: errlog,
 	}
 
-	var g run.Group
-	g.Add(func() error {
-		return httpSrv.Serve(listener)
-	}, func(error) {
+	go func() {
+		<-ctx.Done()
 		runutil.CloseWithLogOnErr(h.logger, listener, "receive HTTP listener")
```

Comment:
Why not having `handler.Close` method and closing listener there? And having `Close()` method invoked in root `oklog/run`? Would be more explicit and consistent? (:

Reply:
yes absolutely sounds good, I assumed we were using context for consistency :)

Reply:
Hard to say about any consistency here.. but you are right we mix things everywhere in this space. (: We can do both, but this sounds like extra hidden go routine (: